java-checks/src/main/java/org/sonar/java/checks/helpers/UnitTestUtils.java

```diff
@@ -42,6 +44,13 @@ public final class UnitTestUtils {
       "|laxCheckpoint|succeedingThenComplete");
   private static final Pattern TEST_METHODS_PATTERN = Pattern.compile("test.*|.*Test");
 
+  private static final List<String> ASSERTJ_ASSERTION_METHODS_PREFIXES = List.of("has", "doesNot", "is", "contains", "assert");
+
+  @VisibleForTesting
+  static final Pattern ASSERTJ_ASSERTION_METHODS_PATTERN = Pattern.compile(
+    String.join("|", "returns", "contains", "allMatch", "is", "has", "satisfies") + "|" +
+      "((" + String.join("|", ASSERTJ_ASSERTION_METHODS_PREFIXES) + ")" + "[A-Z].*)");
```

Comment:
I don't see the advantage of this indirection `String.join("|", "a", "b"))` instead of just `"a|b"`. And I think we could treat all those elements as `method name` or `method name prefix`, I don't have in mind a false-negative if we do so. So I suggest:
```suggestion
  @VisibleForTesting
  static final Pattern ASSERTJ_ASSERTION_METHODS_PATTERN = Pattern.compile(
    "(allMatch|assert|contains|doesNot|has|is|returns|satisfies)([A-Z].*)?");
```